choreolib/src/main/java/choreo/auto/AutoRoutine.java

```diff
@@ -45,31 +48,22 @@ public class AutoRoutine {
   /** The amount of times the routine has been polled */
   protected int pollCount = 0;
 
-  /** Returns true if the alliance is known or is irrelevant (i.e. flipping is not being done) */
-  protected BooleanSupplier allianceKnownOrIgnored = () -> true;
-
   /**
    * A constructor to be used when inhereting this class to instantiate a custom inner loop
    *
    * @param factory The factory that created this loop
    * @param name The name of the loop
    * @param loop The inner {@link EventLoop}
+   * @param allianceKnownOrIgnored Returns true if the alliance is known or is irrelevant (i.e.
+   *     flipping is not being done).
+   * @see AutoFactory#newRoutine Creating a loop from a AutoFactory
    */
-  protected AutoRoutine(AutoFactory factory, String name, EventLoop loop) {
+  protected AutoRoutine(
+      AutoFactory factory, String name, EventLoop loop, BooleanSupplier allianceKnownOrIgnored) {
```

Comment:
`AutoRoutine` is initialized in two places: as `VOID-ROUTINE`, and in `newRoutine()`. In both cases, they effectively use `new EventLoop()`. `AutoRoutine.loop` should simply be initialized next to its declaration, and this constructor should be removed.